### PR TITLE
add a dev/dartdoc.dart script to generate docs for the packages/ packages

### PR DESCRIPTION
**Pull request by devoncarew**
- add a dev/dartdoc.dart script to generate docs for the packages/ packages (in progress for https://github.com/flutter/flutter/issues/2399)

This isn't plumbed in yet - it requires https://github.com/dart-lang/dartdoc/pull/1132 landing and dartdoc being re-published.

### CODE REVIEW

**Comment by eseidelGoogle** [comment]
You may want README.md instead of readme.md.

**Comment by devoncarew** [comment]
I can change it; I often put in readme because README is just so loud.

**Comment by eseidelGoogle** [comment]
Doesn't matter. :)  My comment was just noise.  carry on.

**Comment by abarth** [comment]
Is this `doc` directory something we expect end-developers to care about or is it an implementation detail of how we generate the docs.flutter.io web site?  If its an implementation detail, we might want to put it in the `dev` directory.

**Comment by devoncarew** [comment]
Just an implementation detail. It's an existing dir happy to move it around. I think moves will impact the infra bots.

**Comment by abarth** [comment]
Feel free to move the dir in a separate patch if you like.

**Comment by abarth** [comment]
LGTM

**Comment by devoncarew** [comment]
I copied the contents of the `doc/` dir to `dev/docs/`. I'll remove the older copies once infra had been updated.

This works with the just published `0.9.3` version of dartdoc (but will be slightly happier with the soon to be published `0.9.3+1`).

**Comment by eseidelGoogle** [comment]
FYI, Git supports symlinks if you need to in the future.

File: packages/flutter_tools/pubspec.yaml [hunk 1]
```diff
@@ -41,3 +41,7 @@ dependencies:
 
 dev_dependencies:
   mockito: ^0.11.0
+
+# Exclude this package from the hosted API docs.
+dartdoc:
+  nodoc: true
```
Comment (Hixie) [inline packages/flutter_tools/pubspec.yaml]: we should exclude these from the count in `flutter analyze` also, then

File: packages/flutter_markdown/lib/flutter_markdown.dart [hunk 2]
```diff
@@ -2,6 +2,7 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+/// A library to render markdown formatted text.
```
Comment (Hixie) [inline packages/flutter_markdown/lib/flutter_markdown.dart]: @vlidholt Could you add some more documentation to this comment (like how to use the library)? No urgent need to go and dart-doc the whole library but if we could get a few paragraphs of introductory material here that would go a long way for now.

**Pull request merged**